### PR TITLE
KFSPTS-14681 Add RASS Award handling

### PR DESCRIPTION
**Pull request by cah292**
This PR adds the ability to create KFS Awards from RASS XML. It also adds some related unit tests, and adds some Proposal-related handling to the unit tests that did not get added with the earlier RASS Proposal PR.

Note that some significant updates were needed to get this working. I added some new helper objects and infrastructure to allow for properly handling edits to certain Award sub-objects (such as project directors). I also had to change the point at which new transactions start when creating new objects, and also renamed and refactored the related services as a result.

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/rass/batch/RassObjectTranslationDefinition.java [hunk 1]
```diff
@@ -28,6 +29,12 @@ public void setRootXmlObjectListPropertyName(String rootXmlObjectListPropertyNam
         this.rootXmlObjectListPropertyName = rootXmlObjectListPropertyName;
     }
 
+    public void makeBusinessObjectActiveIfApplicable(T xmlObject, R newBusinessObject) {
```
[the remark this answers is not in the document]
Comment (ajd299) [reply src/main/java/edu/cornell/kfs/rass/batch/RassObjectTranslationDefinition.java]: - Is the unused xmlObject parameter necessary?
- If not, we could change it to below

`makeBusinessObjectActiveIfApplicable(PersistableBusinessObject newBusinessObject) {...`

File: src/main/java/edu/cornell/kfs/rass/batch/RassValueConverterBase.java [hunk 2]
```diff
@@ -1,9 +1,65 @@
...
+    public void setDataDictionaryService(DataDictionaryService dataDictionaryService) {
+        this.dataDictionaryService = dataDictionaryService;
+    }
+
+    @Override
+    public Object convert(Class<? extends PersistableBusinessObject> businessObjectClass, String propertyName, Object propertyValue) {
+        return cleanSimplePropertyValue(businessObjectClass, propertyName, propertyValue);
+    }
+
+    protected Object cleanSimplePropertyValue(
```
[the remark this answers is not in the document]
Comment (ajd299) [reply src/main/java/edu/cornell/kfs/rass/batch/RassValueConverterBase.java]: I've added debug statements for null value and unhandled instance type.

File: src/main/java/edu/cornell/kfs/rass/batch/RassValueConverterBase.java [hunk 3]
```diff
@@ -1,9 +1,65 @@
...
+        } else if (propertyValue instanceof Boolean) {
+            return cleanBooleanValue(businessObjectClass, propertyName, (Boolean) propertyValue);
+        } else {
+            return propertyValue;
+        }
+    }
+
+    protected String cleanStringValue(
+            Class<? extends PersistableBusinessObject> businessObjectClass, String propertyName, String propertyValue) {
+        String cleanedValue = StringUtils.defaultIfBlank(propertyValue, null);
```
[the remark this answers is not in the document]
Comment (ajd299) [reply src/main/java/edu/cornell/kfs/rass/batch/RassValueConverterBase.java]: I've added an if statement to trim if not null

File: src/main/java/edu/cornell/kfs/rass/batch/RassListPropertyDefinition.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package edu.cornell.kfs.rass.batch;
+
+import java.util.Map;
+
+public class RassListPropertyDefinition extends RassPropertyDefinition {
```
[the remark this answers is not in the document]
Comment (ajd299) [reply src/main/java/edu/cornell/kfs/rass/batch/RassListPropertyDefinition.java]: I do not have a better name

[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/rass/batch/RassListPropertyDefinition.java]: OK, lets leave as is

File: src/main/java/edu/cornell/kfs/rass/batch/RassObjectTranslationDefinition.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/rass/batch/RassObjectTranslationDefinition.java]: I think Chad designed this class to be easily extended, so we want to keep both input parameters using T and R and the type on variables.  The "<T extends RassXmlObject, R extends PersistableBusinessObject> " between public and void make it more understandable for whomever is using the code in the future.

**Pull request merged**